Default boxplot style rebase

jenshnielsen opened the pull request:
Rebase of #5523 on master hopefully fixing the travis issues.

jenshnielsen commented:
@phobson I hope you are ok with me making a few changes on top of your PR so we can get it merged before we cut the 2.x beta. I have tried to do a few modifications as possible to restore old default behaviours with matplotlib 1.5.x classic style and fix the tests. Let me know if you can spot any issues. 
Otherwise I think this is ready for review. I will need to squash all image changes together before merging  but prefer that to wait until this has been reviewed.

phobson commented:
@jenshnielsen absolutely! sorry i was lagging behind

tacaswell commented on lib/matplotlib/rcsetup.py:
```diff
@@ -941,33 +941,37 @@ def validate_animation_writer_path(p):
     'boxplot.showfliers': [True, validate_bool],
     'boxplot.meanline': [False, validate_bool],
 
-    'boxplot.flierprops.color': ['C0', validate_color],
```
All of these changes need to be reflected into the template.

jenshnielsen replied:
As far as I can see they already do

tacaswell commented:
Something is not right here, the output from http://matplotlib.org/examples/pylab_examples/boxplot_demo.html does not seem to respect the color (either on this branch or v1.5.1) but does work on master http://matplotlib.org/devdocs/examples/pylab_examples/boxplot_demo.html

phobson commented:
@tacaswell that is weird. I'm not sure what's going on.

@jenshnielsen I'm not sure why the medians aren't showing up in this example either: https://github.com/matplotlib/matplotlib/pull/6481/files#diff-b5878226d9a3230ab74d34de5a0f994f

jenshnielsen commented:
@tacaswell I fixed the color issue (The flier color was set by the marker argument but not the face and edgecolor which was set to default values)

jenshnielsen commented:
The effect of the color can also be seen in the changes to boxplot_mod_artists_after_plotting.png
The fliers used to have their color set from 
```
bp = ax.boxplot(x, sym="o")
    for key in bp:
        for obj in bp[key]:
            obj.set_color('green')
```
This is no longer the case because edge and facecolor now takes effect

jenshnielsen commented:
Running out of time to look more at this at the moment. I will try to get back to it later today.

This will need some test image updated after the fix above.

There seems to be something wrong with test_bxp_custompatchartist
the linestyle is not correctly reflected in the test image (dotted)

jenshnielsen commented:
@phobson Sorry for being stupid but can you point me exactly to which example is missing the median? 
I can only see bxp_patchartist.png which simply has the same patch color as the plotted under the median making it hard to see

phobson commented:
@jenshnielsen happily. It's right here: https://github.com/matplotlib/matplotlib/pull/6481/files#diff-b5878226d9a3230ab74d34de5a0f994f

oddly, the custom patch artist example is fine.
https://github.com/matplotlib/matplotlib/pull/6481/files#diff-ac6c423875c2703c15afceabfdaf3692

jenshnielsen commented:
So we are talkign about bxp_patchartist.png? 
In that test we are plotting a median which has the color C0 on top of a patch with a color C0 so naturally it's not really visible.

We should probably either change the default median color or change the default patch color to use some thing else than `patch.facecolor`

phobson commented:
Oh yeah. That makes sense. This is tricky. My preference is that if we're filling in the box with a patch artist, the the median should be the same color as the border of the patch artist. 

--
Sorry if this was unintelligible. I'm on my phone.




On Mon, May 30, 2016 at 7:48 AM -0700, "Jens Hedegaard Nielsen" <notifications@github.com> wrote:












So we are talkign about bxp_patchartist.png? 

In that test we are plotting a median which has the color C0 on top of a patch with a color C0 so naturally it's not really visible.



We should probably either change the default median color or change the default patch color to use some thing else than patch.facecolor



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

jenshnielsen commented:
Thats not really easy to do unfortunately. To do that I think we would need to add a different rc param for the median color when using patch. I changed the defaults mean and median to C1 and C2 which are different from the patch color. I squashed the image changing commits together

tacaswell commented:
backported to v2.x as ff98856a15cbd236f0139454c9bddef1dad0294d

phobson commented:
thanks @tacaswell and @jenshnielsen !!
